api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -918,6 +950,45 @@ type AzureNodePoolPlatform struct {
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
 }
 
+// AzureImageInfo is the id of the image to boot from
+type AzureImageInfo struct {
+	ID *string `json:"imageID,omitempty"`
+}
+
+// MarketplaceImage specifies the information needed to create an Azure VM from an Azure Marketplace image. This struct
+// replicates the same fields found in CAPZ - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go.
+type MarketplaceImage struct {
+	// Publisher is the name of the organization that created the image
+	//
+	// +kubebuilder:validation:MinLength=1
+	Publisher string `json:"publisher,omitempty"`
```

Comment:
`+kubebuilder:validation:MinLength=1` means this is required, right? we don't need `omitempty` then.

Reply:
same comment for the rest of the fields in this struct.